ipie/walkers/pop_controller.py

```diff
@@ -138,7 +138,7 @@ def get_buffer(walkers, iw):
     buff = xp.zeros(walkers.buff_size, dtype=numpy.complex128)
     for d in walkers.buff_names:
         data = walkers.__dict__[d]
-        if data is None:
+        if (data is None) or isinstance(data, (int, float, complex, numpy.float64, numpy.complex128)):
```

Comment:
Create ThermalPopController by inheriting from PopController. Localize the changes into addons so that the main pop_controller.py does not depend on addons.